lib/inventory/metadata/metadata.go

```diff
@@ -0,0 +1,322 @@
+}
+
+// setDefaults sets the values of several methods used to read files, execute
+// commands, performing http requests, etc.
+// Having these methods configurable allows us to mock them in tests.
+func (c *fetchConfig) setDefaults() {
+	if c.context == nil {
+		c.context = context.Background()
+	}
+	if c.getenv == nil {
+		c.getenv = os.Getenv
+	}
+	if c.readFile == nil {
+		c.readFile = os.ReadFile
+	}
+	if c.execCommand == nil {
+		c.execCommand = func(name string, args ...string) ([]byte, error) {
+			return exec.Command(name, args...).Output()
+		}
+	}
+	if c.httpDo == nil {
+		c.httpDo = func(req *http.Request) (*http.Response, error) {
+			client, err := defaults.HTTPClient()
+			if err != nil {
+				return nil, err
+			}
+			client.Timeout = 5 * time.Second
+			return client.Do(req)
+		}
+	}
+	if c.kubeClient == nil {
+		c.kubeClient = getKubeClient()
+	}
+}
+
+// getKubeClient returns a kubernetes client in case the instance is running on
+// kubernetes. It returns nil otherwise.
+func getKubeClient() kubernetes.Interface {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Debugf("Failed to get kubernetes cluster config: %s", err)
+		return nil
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Debugf("Failed to create kubernetes client: %s", err)
+		return nil
+	}
+	return client
+}
+
+// fetch fetches all metadata.
+func (c *fetchConfig) fetch() *Metadata {
+	return &Metadata{
+		OS:                    c.fetchOS(),
+		OSVersion:             c.fetchOSVersion(),
+		HostArchitecture:      c.fetchHostArchitecture(),
+		GlibcVersion:          c.fetchGlibcVersion(),
+		InstallMethods:        c.fetchInstallMethods(),
+		ContainerRuntime:      c.fetchContainerRuntime(),
+		ContainerOrchestrator: c.fetchContainerOrchestrator(),
+		CloudEnvironment:      c.fetchCloudEnvironment(),
+	}
+}
+
+// fetchOS returns the value of GOOS.
+func (c *fetchConfig) fetchOS() string {
+	return runtime.GOOS
+}
+
+// fetchHostArchitecture returns the output of arch.
+func (c *fetchConfig) fetchHostArchitecture() string {
+	arch, err := c.exec("arch")
+	if err != nil {
+		return ""
+	}
+
+	return arch
+}
+
+// fetchInstallMethods returns the list of methods used to install the instance.
+func (c *fetchConfig) fetchInstallMethods() []string {
+	installMethods := []string{}
+	if c.dockerfileInstallMethod() {
+		installMethods = append(installMethods, "dockerfile")
+	}
+	if c.helmKubeAgentInstallMethod() {
+		installMethods = append(installMethods, "helm_kube_agent")
+	}
+	if c.nodeScriptInstallMethod() {
+		installMethods = append(installMethods, "node_script")
+	}
+	if c.systemctlInstallMethod() {
+		installMethods = append(installMethods, "systemctl")
+	}
+	return installMethods
+}
+
+// dockerfileInstallMethod returns true if the instance was installed using our
+// Dockerfile.
+func (c *fetchConfig) dockerfileInstallMethod() bool {
+	return c.getenv("TELEPORT_INSTALL_METHOD_DOCKERFILE") == "true"
```

Comment:
```suggestion
	return c.getenv("TELEPORT_INSTALL_METHOD_DOCKERFILE") != ""
```
or maybe something fancier with `utils.ParseBool`, allowing an explicit `0` or `false` while letting everything else count as true?

Reply:
Done in bff342e63298429a393f729455a6ae624dc763bc.